src/figuresClasses.ts

```diff
@@ -1,11 +1,86 @@
-export interface Figure {}
+type ValidShapes = 'triangle' | 'circle' | 'rectangle';
+type ValidColors = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ValidShapes;
+  color: ValidColors;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  public shape: ValidShapes = 'triangle';
+
+  constructor(
+    public color: ValidColors,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to explicitly define them again inside the constructor body. The constructor can be empty in this case.